test/test_basic.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python3
+
+import base64
+import unittest
+import subprocess
+import os
+import time
+import urllib.error
+import urllib.request
+import ssl
+import json
+
+
+dirname = os.path.dirname(__file__)
+projroot = os.path.join(dirname, "..")
+
+
+class IntegrationTest(unittest.TestCase):
+
+    def setUp(self):
+        subprocess.check_call(["make", "containers"], cwd=projroot)
+        subprocess.check_call(["make", "run"], cwd=projroot)
+
+    def tearDown(self):
+        # Make this only on ubuntu-22.04
+        version = subprocess.check_output(["podman", "-v"]).decode().split(' ')[-1]
+        if version[0] == '4':
+            subprocess.check_call(["make", "clean"], cwd=projroot)
+        else:
+            subprocess.check_call(["make", "clean-test"], cwd=projroot)
+
+    def request(self, url, retries=20):
+        context = ssl.create_default_context(cafile=os.path.join(projroot, '3scale', 'certs', 'ca.crt'))
+        context.check_hostname = False
+        b64 = base64.b64encode(b'admin:foobar').decode()
+
+        request = urllib.request.Request(url)
+        request.add_header('Authorization', f'Basic {b64}')
+        tries = 0
+
+        while tries <= retries:
+            try:
+                response = urllib.request.urlopen(request, context=context, timeout=1)
+                if response.status >= 200 and response.status < 300:
+                    return response
+            except urllib.error.HTTPError as exc:
+                if 'Bad Gateway' in str(exc):
+                    pass
```

Comment:
No-op -- if the `if` condition is false, this also behaves like `pass`. Perhaps you mean `else: raise`?